### PR TITLE
Protect map from concurrent access (Fixes #2915)

### LINKED ISSUE #2915: [distributed] Race in PeersRPCServer() usage sporadically reproducible.
```
==================
WARNING: DATA RACE
Write at 0x000001739120 by goroutine 36:
  github.com/minio/minio/cmd.initConfig()
      /home/travis/gopath/src/github.com/minio/minio/cmd/config-v9.go:80 +0xc14
  github.com/minio/minio/cmd.newTestConfig()
      /home/travis/gopath/src/github.com/minio/minio/cmd/test-utils_test.go:362 +0xdf
  github.com/minio/minio/cmd.StartTestPeersRPCServer()
      /home/travis/gopath/src/github.com/minio/minio/cmd/test-utils_test.go:249 +0xbb
  github.com/minio/minio/cmd.(*TestPeerRPCServerData).Setup()
      /home/travis/gopath/src/github.com/minio/minio/cmd/event-notifier_test.go:164 +0x9f
  github.com/minio/minio/cmd.TestAddRemoveBucketListenerConfig()
      /home/travis/gopath/src/github.com/minio/minio/cmd/event-notifier_test.go:419 +0xd8
  testing.tRunner()
      /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/testing/testing.go:610 +0xc9
Previous read at 0x000001739120 by goroutine 155:
  [failed to restore the stack]
Goroutine 36 (running) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/testing/testing.go:646 +0x52f
  testing.RunTests.func1()
      /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/testing/testing.go:793 +0xb9
  testing.tRunner()
      /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/testing/testing.go:610 +0xc9
  testing.RunTests()
      /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/testing/testing.go:799 +0x4ba
  testing.(*M).Run()
      /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/testing/testing.go:743 +0x12f
  main.main()
      github.com/minio/minio/cmd/_test/_testmain.go:788 +0x1b8
Goroutine 155 (running) created at:
  net/rpc.(*Server).ServeCodec()
      /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/net/rpc/server.go:477 +0x4ba
  net/rpc.(*Server).ServeConn()
      /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/net/rpc/server.go:454 +0x64a
  net/rpc.(*Server).ServeHTTP()
      /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/net/rpc/server.go:696 +0x561
  github.com/minio/minio/vendor/github.com/gorilla/mux.(*Router).ServeHTTP()
      /home/travis/gopath/src/github.com/minio/minio/vendor/github.com/gorilla/mux/mux.go:107 +0x184
  net/http.serverHandler.ServeHTTP()
      /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/net/http/server.go:2202 +0xbb
  net/http.(*conn).serve()
      /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/net/http/server.go:1579 +0x5f6
```

### PR DESCRIPTION
**Pull request by donatello**
Protects the Peers RPC clients map from concurrent access to fix a
data race condition.

### CODE REVIEW

File: cmd/s3-peer-client.go [hunk 1]
```diff
@@ -29,14 +30,16 @@ type s3Peers struct {
 	// client connections
 	rpcClient map[string]*AuthRPCClient
```
Comment (harshavardhana) [inline cmd/s3-peer-client.go]: This variable should be 'rpcClients'

File: cmd/s3-peer-client.go [hunk 2]
```diff
@@ -55,15 +58,22 @@ func (s3p *s3Peers) GetPeers() []string {
 }
 
 func (s3p *s3Peers) GetPeerClient(peer string) *AuthRPCClient {
+	// take a read lock
```
Comment (harshavardhana) [inline cmd/s3-peer-client.go]: Capitalize here.. like `// Take a read lock`

File: cmd/s3-peer-client.go (hunk 1 above)
Comment (donatello) [reply cmd/s3-peer-client.go]: Done

File: cmd/s3-peer-client.go (hunk 2 above)
Comment (donatello) [reply cmd/s3-peer-client.go]: Done.

**Pull request merged**